### DISCUSSION TITLE
Iterate through pages on REST queries with limits.

### DISCUSSION THREAD

**Discussion by tokkisva** [Budibase/budibase#8720] [Help] [answered]
Hi all,

The REST API I am working with has a max record limit of 250 results per query, but does allow pagination.

Query looks like this:

GET {url}/search?{filters} & limit 250 & page 1

While I am able to cycle through the pages in the table from the GUI using the pagination tab- what I am trying to do is pull **all** pages from the API that match the filtered query and display them in a table. So if there are 4 pages of data, I would see 1000 records in my table.

I'm hoping budibase can do something like this:

EXAMPLE

GET {url}/search?{filters} & limit 250 & page 1
GET {url}/search?{filters} & limit 250 & page 2
GET {url}/search?{filters} & limit 250 & page 3
GET {url}/search?{filters} & limit 250 & page 4
GET {url}/search?{filters} & limit 250 & page 5
GET {url}/search?{filters} & limit 250 & page 6
... etc

and then display all of the results from the queries in a table. In the example above, my table would have 1500 results.

Is this possible in budibase? Not really sure where to start.. I thought transformer could do this but apparently it cannot according to this thread: https://github.com/Budibase/budibase/discussions/3875. 

Thanks in advance!

**Comment by melohagan** [comment, marked as answer]
Hey @tokkisva 

Budibase does support pagination, however it'll simply match the API, so in this case it would only show 250 records: https://docs.budibase.com/docs/rest-pagination

It's a bit ugly, but you could add 6 data providers to your screen, and then add repeaters to display the row data.

If you really want to use a single table component, then you may need to spin up your own lambda or microservice that would aggregate the six calls into one response, that could then be used within Budibase.
Alternatively there's custom datasources: https://docs.budibase.com/docs/custom-datasource
You could build a simple custom datasource that aggregates your six calls as well.

**Reply by tokkisva** [reply]
Hi @melohagan 

I see, i'll look into getting lambda setup because this seems like the way to go. Apologies if this is a dumb question, I'm not familiar with lambda at all; would it be setup as its own REST data source then? If so that would be perfect- since I am looking to use a binding from the form and then transform the data retrieved from that 'batched' JSON response!



Thanks!

**Reply by melohagan** [reply]
An AWS lambda would return a response like any other REST API, so you would use the REST datasource to pull that aggregated response. 
As I mentioned, a custom datasource could be another option if you don't want to go down the lambda route.

**Reply by tokkisva** [reply]
Solid, going to review options and hit the books. Really appreciate your help here melohagan!

**Reply by lgasteceau** [reply]
Hello,

I developed a shorter version of @melohagan idea : instead of creating X data providers for the X pages, I created a "pagination" BB Table with 1 column "pageNumber" and simply add as many as rows you need with 1, 2, 3, 4, 5, 6, ...

Then in your page, create a data provider connected to pageNumber table, and a repeater attached to it.

You can insert your REST data provider inside the pagination repeater. It will iterate over your pages passing as a binding the repeater.pageNumber.

I recomand adding a container inside your query data provider, with a condition "hide component if dataProvider.rows.length = 0" so you don't have blank results for the remaining pages.
